### PR TITLE
[Dumboiroy] iP

### PR DESCRIPTION
**Pull request by Dumboiroy**
# Seeyes 🎉

> This is a **minimal** PR description with all *required* GFMD elements.

- Todo
- Deadline
- Event
Create your first event with: `todo <taskname>`

## Features
- [x] ~~Working chatbot~~
- [ ] GUI interface  
- [ ] Advanced features

### Changes
1. Added `Seeyes` class
2. Implemented task management
3. Created **command** system

```java
public static void main(String[] args) {
    new Seeyes("./data/data.txt").run();
}
```
Check out the [CS2103T website](https://nus-cs2103-ay2526s1.github.io/) for more info.

**Bold**, *italic*, and <del>strikethrough</del> text formatting included.

### CODE REVIEW

File: src/test/java/seeyes/command/AddTaskCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        // Act
+        CommandResult result = addTaskCommand.execute();
+
+        // Assert
+        assertTrue(result.getMessage().contains("Successfully added:"));
+        assertTrue(result.getMessage().contains("Buy groceries"));
+        assertTrue(result.getMessage().contains("[T]"));
+        assertEquals(1, taskList.size()); // Task should be added to list
+        assertEquals(todoTask, taskList.getTaskByIndex(0)); // Verify the exact
+                                                            // task was added
```
Comment (christopherchong) [inline src/test/java/seeyes/command/AddTaskCommandTest.java]: Maybe the comment can be on a single line?

File: src/test/java/seeyes/command/AddTaskCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        CommandResult result = addTaskCommand.execute();
+
+        // Assert
+        assertTrue(result.getMessage().contains("Successfully added:"));
+        assertTrue(result.getMessage().contains("Project Meeting"));
+        assertTrue(result.getMessage().contains("[E]"));
+        assertEquals(1, taskList.size()); // Task should be added to list
+        assertEquals(eventTask, taskList.getTaskByIndex(0)); // Verify the
+                                                             // exact task
+                                                             // was added
```
Comment (christopherchong) [inline src/test/java/seeyes/command/AddTaskCommandTest.java]: Maybe the comment can be on a single line?

File: src/test/java/seeyes/command/AddTaskCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        CommandResult result = addTaskCommand.execute();
+
+        // Assert
+        assertTrue(result.getMessage().contains("Successfully added:"));
+        assertTrue(result.getMessage().contains("Submit assignment"));
+        assertTrue(result.getMessage().contains("[D]"));
+        assertEquals(1, taskList.size()); // Task should be added to list
+        assertEquals(deadlineTask, taskList.getTaskByIndex(0)); // Verify the
+                                                                // exact task
+                                                                // was added
```
Comment (christopherchong) [inline src/test/java/seeyes/command/AddTaskCommandTest.java]: Maybe the comment can be on a single line?

File: src/test/java/seeyes/task/TaskTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import seeyes.exception.InvalidTaskTypeException;
+
+public class TaskTest {
+    @Test
+    public void fromStringTestInvalidTaskType() {
+        // Test invalid task type "XX"
+        String taskString = "XX|0|Invalid task";
+
+        InvalidTaskTypeException exception = assertThrows(InvalidTaskTypeException.class,
+                () -> Task.fromString(taskString));
```
Comment (christopherchong) [inline src/test/java/seeyes/task/TaskTest.java]: Perhaps the parentheses should be on the previous line?

File: src/test/java/seeyes/task/TaskTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+        // Verify the exception message contains the invalid task type
+        assertTrue(exception.getMessage().contains("XX"));
+        assertTrue(exception.getMessage().contains("does not exist"));
+
+        // Test another invalid task type "ZZ"
+        String taskString2 = "ZZ|1|Another invalid task";
+
+        InvalidTaskTypeException exception2 = assertThrows(InvalidTaskTypeException.class,
+                () -> Task.fromString(taskString2));
```
Comment (christopherchong) [inline src/test/java/seeyes/task/TaskTest.java]: Perhaps the parentheses should be on the previous line?

File: src/test/java/seeyes/task/TaskTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        assertTrue(task4 instanceof TodoTask);
+        assertTrue(task4.toString().contains("Task with special chars!@#$%"));
+
+        // Test boundary value for done status (non-1 should be false)
+        String taskString5 = "TD|2|Task with status 2";
+        Task task5 = Task.fromString(taskString5);
+
+        assertTrue(task5 instanceof TodoTask);
+        assertFalse(task5.toString().contains("[X]")); // Should be false for
+                                                       // non-1 values
```
Comment (christopherchong) [inline src/test/java/seeyes/task/TaskTest.java]: Maybe the comment can be on a single line?

File: src/main/java/seeyes/command/DeleteCommand.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
+package seeyes.command;
+
+import seeyes.exception.InvalidCommandException;
+
+/**
+ * Command to delete a task from the task list.
+ */
+public class DeleteCommand extends Command {
+    private int indexToMark;
```
Comment (christopherchong) [inline src/main/java/seeyes/command/DeleteCommand.java:9]: Clear variable naming!

File: src/main/java/seeyes/Seeyes.java
Comment (christopherchong) [inline src/main/java/seeyes/Seeyes.java:1]: The project follows good naming standards for classes, methods and variables. Good job! 👍

File: src/main/java/seeyes/parser/Parser.java [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        /**
+         * Extracts a CommandType from a given command string.
+         *
+         * @param commandString
+         *            the command string
+         * @return the corresponding CommandType
+         * @throws InvalidCommandException
+         *             if the command is not recognized
+         */
+        public static CommandType fromString(String commandString) throws InvalidCommandException {
```
Comment (christopherchong) [inline src/main/java/seeyes/parser/Parser.java]: Perhaps this method name could be renamed to use a verb like 'getCommandFromString' for example?

File: src/main/java/seeyes/storage/Storage.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    /**
+     * Loads tasks from the storage file.
+     *
+     * @return the loaded task list
+     * @throws StorageException
+     *             if loading fails
+     */
+    public TaskList load() throws StorageException {
```
Comment (christopherchong) [inline src/main/java/seeyes/storage/Storage.java:43]: Maybe method name can be more informative, such as loadFile()?

File: src/main/java/seeyes/storage/Storage.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Saves the task list to the storage file.
+     *
+     * @param taskList
+     *            the task list to save
+     * @return a success message
+     * @throws StorageException
+     *             if saving fails
+     */
+    public String save(TaskList taskList) throws StorageException {
```
Comment (christopherchong) [inline src/main/java/seeyes/storage/Storage.java:82]: Maybe method name can be more informative, such as saveToFile()?

File: src/main/java/seeyes/task/Task.java [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+        /**
+         * Converts a string to a TaskType.
+         *
+         * @param taskString
+         *            the string to convert
+         * @return the corresponding TaskType
+         * @throws InvalidTaskTypeException
+         *             if the task type is invalid
+         */
+        public static TaskType fromString(String taskString) throws InvalidTaskTypeException {
```
Comment (christopherchong) [inline src/main/java/seeyes/task/Task.java]: Perhaps the method name can be renamed to include a verb?

File: src/main/java/seeyes/task/Task.java [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    /**
+     * Creates a task from a saved string representation.
+     *
+     * @param taskString
+     *            the string representation from file
+     * @return the reconstructed task
+     */
+    public static Task fromString(String taskString) {
```
Comment (christopherchong) [inline src/main/java/seeyes/task/Task.java]: Perhaps the method name can be renamed to include a verb?

**Review by christopherchong — COMMENTED** [review]
LGTM. The project uses really good naming. Just some minor nits to fix.

File: src/main/java/seeyes/parser/Parser.java [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+     * @return array of parsed parameters
+     * @throws InvalidCommandException
+     *             if parameters are invalid
+     */
+    private static String[] parseTaskParams(CommandType taskType, String paramString) throws InvalidCommandException {
+        String[] params;
+        switch (taskType) {
+        case TODO:
+        case FIND:
+            return new String[] { paramString };
```
Comment (yhthia) [inline src/main/java/seeyes/parser/Parser.java:190]: Perhaps use Egyptian style brackets for this?

File: src/main/java/seeyes/ui/Ui.java [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+ * <p>
+ * Usage example:
+ *
+ * <pre>
+ * Ui ui = Ui.getUi();
+ * String command = ui.getNextUserCommand();
+ * ui.say("Hello!");
+ * </pre>
+ */
+public class Ui {
```
Comment (yhthia) [inline src/main/java/seeyes/ui/Ui.java:35]: I like that the method names and Javadoc comments are concise and descriptive👍

File: src/test/java/seeyes/command/AddTaskCommandTest.java [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+        // Act
+        CommandResult result = addTaskCommand.execute();
+
+        // Assert
+        assertTrue(result.getMessage().contains("Successfully added:"));
+        assertTrue(result.getMessage().contains("Buy groceries"));
+        assertTrue(result.getMessage().contains("[T]"));
+        assertEquals(1, taskList.size()); // Task should be added to list
+        assertEquals(todoTask, taskList.getTaskByIndex(0)); // Verify the exact
```
Comment (yhthia) [inline src/test/java/seeyes/command/AddTaskCommandTest.java]: I agree with the other reviewer, perhaps these comments can be written in one line?

File: src/main/java/seeyes/command/UnmarkCommand.java [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @return the result of the command execution
+     * @throws InvalidCommandException
+     *             if the task index is invalid
+     */
+    @Override
+    public CommandResult execute() throws InvalidCommandException {
+        try {
+            taskList.getTaskByIndex(indexToMark).markAsNotDone();
+            return new CommandResult("Bummer. Alright, I've unchecked:\n" + taskList.getTaskByIndex(indexToMark)
```
Comment (yhthia) [inline src/main/java/seeyes/command/UnmarkCommand.java:32]: Good job on using line breaks to make the code more readable 👍

**Review by yhthia — COMMENTED** [review]
LGTM, really clean code overall, just a few minor nits to look through!
